src/tequila/objective/objective.py

```diff
@@ -676,7 +676,8 @@ def extract_variables(self):
             else:
                 variables += []
 
-        return list(set(variables))
+        # Casting to set ensures unique elements but not the order of the elements, so they must be sorted
+        return sorted(list(set(variables)))
```

Comment:
won't work anymore if lt operator is removed. Won't work in general even with lt operator (see comment there).
Currently this won't be noticed since VectorObjective is barely used